llvm/lib/Target/RISCV/MCTargetDesc/RISCVMatInt.cpp

```diff
@@ -57,6 +61,25 @@ static void generateInstSeqImpl(int64_t Val, const MCSubtargetInfo &STI,
     return;
   }
 
+  if (!IsRV64 && STI.hasFeature(RISCV::FeatureVendorXqcili)) {
+    bool FitsOneStandardInst =
+        ((Val & 0xFFF) == 0) || (((Val + 0x800) & 0xFFFFF000) == 0);
```

Comment:
I think is checking for ADDI and LUI, but I don't understand why there is a `+ 0x800` in the LUI check.